core/http/src/header/proxy_proto.rs

```diff
@@ -0,0 +1,51 @@
+use std::fmt;
+
+use uncased::{UncasedStr, AsUncased};
+
+/// A protocol used to identify a specific protocol forwarded by an HTTP proxy.
+/// Value are case-insensitive.
```

Comment:
I think we need to add some links to where this is used (`Config::proxy_proto_header`, the cookie defaults) and where in Rocket someone might interact with it (as a request guard, `Request::proxy_proto_header()`, context_is_likely_secure()).